samples/word/99-preview-apis/manage-footnotes.yaml

```
@@ -0,0 +1,266 @@
+id: word-manage-footnotes
+name: Footnotes
+description: 'This sample shows how to perform basic footnotes operations, including insert, get, and delete.'
+host: WORD
+api_set:
+    WordApi: '1.4'
+script:
+    content: |
+        $("#insert-footnote").click(() => tryCatch(insertFootnote));
+        $("#get-reference").click(() => tryCatch(getReference));
+        $("#get-footnote-type").click(() => tryCatch(getFootnoteType));
+        $("#get-footnote-body").click(() => tryCatch(getFootnoteBody));
+        $("#get-next-footnote").click(() => tryCatch(getNextFootnote));
+        $("#delete-footnote").click(() => tryCatch(deleteFootnote));
+        $("#get-first-footnote").click(() => tryCatch(getFirstFootnote));
+        $("#get-footnotes-from-body").click(() => tryCatch(getFootnotesFromBody));
+        $("#get-footnotes-from-range").click(() => tryCatch(getFootnotesFromRange));
+        $("#setup").click(() => tryCatch(setup));
+
+        /** Set a footnote on the selected content. */
+        async function insertFootnote() {
+          await Word.run(async (context) => {
+            const text = $("#input-footnote").val().toString();
+            const footnote = context.document.getSelection().insertFootnote(text);
+            context.load(footnote);
+            await context.sync();
+
+            console.log("Inserted footnote.");
+          });
+        }
+        /** Get the footnotes in the document body. */
+        async function getFootnotesFromBody() {
+          await Word.run(async (context) => {
+            const body = context.document.body;
+            const result = body.footnotes;
+            result.load("length");
+            await context.sync();
+
+            console.log("Number of footnotes in the document body: " + result.items.length);
+          });
+        }
+        /** Select the footnote's reference mark in the document body. */
+        async function getReference() {
+          await Word.run(async (context) => {
+            const footnotes = context.document.body.footnotes;
+            context.load(footnotes);
+            await context.sync();
+
+            const reference_number = $("#input-reference").val();
+            let mark = (reference_number as number) - 1;
+            const item = footnotes.items[mark];
+            let reference = item.reference;
+            context.load(reference);
+            await context.sync();
+
+            reference.select();
+            await context.sync();
```

This can be written to only load the reference, instead of everything with a footnote. You can load specific properties of items in a collection like this:
```suggestion
      const footnotes = context.document.body.footnotes;
      footnotes.load("items/reference");
      await context.sync();

      const reference_number = $("#input-reference").val();
      let mark = (reference_number as number) - 1;
      const item = footnotes.items[mark];
      let reference = item.reference;
      reference.select();
      await context.sync();
```